### ISSUE TITLE
An opened Dialog component does not disable the UI to which it belongs.

### ISSUE THREAD

**Issue by daniel-scoalainformala** [vaadin/flow#10172]
### Description
An opened Dialog component does not actually disable the UI to which it belongs. The Dialog's parent UI **remains active server-side** and can be interacted with, if the Dialog is removed from the DOM by using DevTools.

### Expected outcome
The parent UI of a Dialog should remain disabled **server-side** while a Dialog is opened, and not be interactive.

### Actual outcome
The parent UI can be enabled client-side by using Browser Developer Tools: delete the Dialog, disable "pointer-events: none" inline style on BODY element

### Live Demo

```
Dialog dialog = new Dialog();
dialog.add("this dialog");
dialog.setCloseOnOutsideClick(false);
dialog.setCloseOnEsc(false);
		
dialog.open();	
```

### Steps to reproduce
1. Create a Dialog
2. Disable closing the dialog on ESC and Outside Click
3. Open the Dialog
4. Enter Developer Tools
5. Delete the Dialog (remove it from the DOM)
6. Disable the "pointer-events: none" inline style from the BODY element

### Browsers Affected
- [x ] Chrome
- [ x] Firefox
- [ x] Safari
- [ x] Edge
- [ x] IE 11
- [ x] iOS Safari
- [ x] Android Chrome

**Comment by pleku** [comment]
Worked on solving this as a part of #7799 and the branch `fix/10172` (commits ^) contains the unfinished changes that would add a new feature to enable supporting adding _modal dialogs_ to the UI so that it will be impossible to user to interact with any other components (or elements) than the ones that are inside the modal dialog. See commit message above for more details. It would also fix #7799 but as there was another related issue (https://github.com/vaadin/vaadin-dialog/issues/229), trying to fix it another (hacky) way instead that address both of those.

**Comment by joheriks** [comment]
Reopening this as #12434 allows for the possibility of server-side modal dialaogs, but this is not (yet) the default behavior. (Currently, you will need to call `ui.addModal(dialog)` prior to `dialog.open()`). Also see #12513 for another caveat.
